weave/Scripts/Player.cs

```diff
@@ -37,11 +37,19 @@ public string PlayerId
     public override void _Ready()
     {
         this.GetNodes();
+        InitiateCurveSpawner();
     }
 
     public override void _PhysicsProcess(double delta)
     {
         Move(delta);
+        CurveSpawner.Step((CircleShape2D) CollisionShape2D.Shape, Rotation, GlobalPosition);
+    }
+
+    private void InitiateCurveSpawner()
```

Comment:
Since the curvespawner is a child to the player now, I think we should add it as a child in the scene and [GetNode]